rclpy/test/test_qos.py

```diff
@@ -124,3 +126,52 @@ def test_preset_profiles(self):
         assert (
             QoSPresetProfiles.SYSTEM_DEFAULT.value ==
             QoSPresetProfiles.get_from_short_key('system_default'))
+
+
+class TestCheckQosCompatibility(unittest.TestCase):
+
+    def test_compatible(self):
+        qos = QoSProfile(
+            depth=1,
+            durability=QoSDurabilityPolicy.VOLATILE,
+            lifespan=Duration(seconds=1),
+            deadline=Duration(seconds=1),
+            liveliness=QoSLivelinessPolicy.AUTOMATIC,
+            liveliness_lease_duration=Duration(seconds=1),
+        )
+        compatibility, reason = qos_check_compatible(
+            qos, qos
+        )
+
+        assert compatibility == QoSCompatibility.OK
+        assert reason == ''
+
+    def test_incompatible(self):
```

Comment:
It might be worth adding a comment that this test is assuming a DDS implementation. It's possible that a "best effort" publisher and "reliable" subscription is a valid match in a non-DDS implementation.